checkov/terraform/graph_builder/foreach/builder.py

```diff
@@ -14,8 +15,14 @@ class ForeachBuilder:
     def __init__(self, local_graph: TerraformLocalGraph):
         self._resource_handler = ForeachResourceHandler(local_graph)
         self._module_handler = ForeachModuleHandler(local_graph)
+        self._data_handler = ForeachDataHandler(local_graph)
 
     def handle(self, foreach_blocks: dict[str, list[int]]) -> None:
+        if self._data_handler.local_graph.enable_datas_foreach_handling:
```

Comment:
can you add some documentation on the order of the calcaulation (data,module, resource).
I think it's very important to document for the future

Reply:
done